From 067bbde1064ed3933785c7f6fa370768151f317f Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Wed, 28 Jan 2015 15:11:03 +0800
Subject: [PATCH] driver: board_hw_info: add the functions to get the main
 board hardware information.

Add the following functions to get the main board hardware information,
 - get_ek_sn(): get the main board's id.
 - get_ek_rev(): get the main board's version.

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 driver/board_hw_info.c  | 10 ++++++++++
 include/board_hw_info.h |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/driver/board_hw_info.c b/driver/board_hw_info.c
index c7a007b..45d6caf 100644
--- a/driver/board_hw_info.c
+++ b/driver/board_hw_info.c
@@ -484,6 +484,16 @@ unsigned int get_dm_sn(void)
 	return (sn >> DM_SN_OFFSET) & SN_MASK;
 }
 
+char get_ek_rev(void)
+{
+	return 'A' + ((rev >> EK_REV_OFFSET) & REV_MASK);
+}
+
+unsigned int get_ek_sn(void)
+{
+	return (sn  >> EK_SN_OFFSET) & SN_MASK;
+}
+
 #if defined(CONFIG_LOAD_ONE_WIRE)
 static unsigned int load_1wire_info(unsigned char *buff, unsigned int size,
 				unsigned int *psn, unsigned int *prev,
diff --git a/include/board_hw_info.h b/include/board_hw_info.h
index b60b685..2aba80f 100644
--- a/include/board_hw_info.h
+++ b/include/board_hw_info.h
@@ -64,6 +64,8 @@ extern unsigned int get_cm_vendor(void);
 extern unsigned int get_dm_sn(void);
 extern unsigned int get_sys_sn(void);
 extern unsigned int get_sys_rev(void);
+extern char get_ek_rev(void);
+extern unsigned int get_ek_sn(void);
 
 extern void load_board_hw_info(void);
 
-- 
1.9.1

